modules/apps/object/object-rest-test/src/testFunctional/tests/FilterCustomObjectsMultipleLevelsRelationshipsWithStringFunctions.testcase

```diff
@@ -196,4 +324,346 @@ definition {
+			var response = ObjectDefinitionAPI.getObjectEntries(
+				en_US_plural_label = "universities",
+				parameter = "filter=contains%28universityStudents%2FsubjectsStudents%2Fname%2C%20%27ath%27%29");
+		}
+
+		task ("Then the response contains the university entry details") {
+			var actual = JSONPathUtil.getProperty(
+				property = "$.items[0].name",
+				response = ${response});
+
+			TestUtils.assertEquals(
+				actual = ${actual},
+				expected = "Liferay");
+
+			HeadlessWebcontentAPI.assertProperNumberOfItems(
+				expectedTotalElement = 1,
+				responseToParse = ${response});
+		}
+	}
+
+	@disable-webdriver = "true"
+	@priority = 4
+	test CanGetCorrectUniversitiesDetailsWhenFilterOverWithCorrectFieldsValueByStartswithFunctions {
+		property portal.acceptance = "true";
+
+		task ("And Given a oneToMany relationship universityStudents") {
+			ObjectDefinitionAPI.createRelationshipWithObjectDefinitionNames(
+				childObjectName = "Student",
+				deletionType = "cascade",
+				name = "universityStudents",
+				parentObjectName = "University",
+				type = "oneToMany");
+		}
+
+		task ("And Given a manyToMany relationship subjectsStudents") {
+			ObjectDefinitionAPI.createRelationshipWithObjectDefinitionNames(
+				childObjectName = "Student",
+				deletionType = "cascade",
+				name = "subjectsStudents",
+				parentObjectName = "Subject",
+				type = "manyToMany");
+		}
+
+		task ("And Given student 'Tom' related to the university 'Liferay' through universities PUT endpoint") {
+			ObjectDefinitionAPI.relateObjectEntriesByName(
+				child_plural_label = "students",
+				childEntryName = "Tom",
+				parent_plural_label = "universities",
+				parentEntryName = "Liferay",
+				relationshipName = "universityStudents");
+		}
+
+		task ("And Given I relate student entry 'Tom' to subject entry 'Math' through subjects PUT endpoint") {
+			ObjectDefinitionAPI.relateObjectEntriesByName(
+				child_plural_label = "students",
+				childEntryName = "Tom",
+				parent_plural_label = "subjects",
+				parentEntryName = "Math",
+				relationshipName = "subjectsStudents");
+		}
+
+		task ("When using get request o/c/universities?filter=startswith(universityStudents/subjectsStudents/name, 'Ma')") {
+			var response = ObjectDefinitionAPI.getObjectEntries(
+				en_US_plural_label = "universities",
+				parameter = "filter=startswith%28universityStudents%2FsubjectsStudents%2Fname%2C%20%27Ma%27%29");
+		}
+
+		task ("Then the response contains the university entry details") {
+			var actual = JSONPathUtil.getProperty(
+				property = "$.items[0].name",
+				response = ${response});
+
+			TestUtils.assertEquals(
+				actual = ${actual},
+				expected = "Liferay");
+
+			HeadlessWebcontentAPI.assertProperNumberOfItems(
+				expectedTotalElement = 1,
+				responseToParse = ${response});
+		}
+	}
+
+	@disable-webdriver = "true"
+	@priority = 3
+	test CanGetNonFilteredItemsWhenFilteringByNonExistentFieldValueByStartswithFunctions {
```

Comment:
this name could be changed as well, according to what we did with `CannotFilterStudentsByStringFunctions`

Reply:
I will modify it as CannotGetFilteredItemsWhenFilteringByNonExistentFieldValueByStartswithFunctions

Reply:
maybe we could follow the same format as `CannotFilterStudentsByStringFunctions`? So it would be `CannotFilterByStringFunctionsWithNonExistentFieldValue`